### PR TITLE
improved build gradle script

### PR DESCRIPTION
**Pull request by OleksandrKucherenko**
- extraction of rootProject properties (according to latest codestyle of RN)
- support of symbolic links as a project directory (live re-mapping of directory structures)

## Motivation (required)

https://github.com/react-native-community/react-native-image-picker/issues/1081

## Test Plan (required)

- connect RN module to your application
- create inside android folder sub-folder `modules`
- create symbolic link to `modules/react-native-image-picker` -> `react-native-image-picker/android`
- add the dependency in an android app project `settings.gradle` as `include ':modules:react-native-image-picker'`
- try to compile the project.

Expected: compilation should pass.
Unexpected: compilation failed with JsonSlupper exception - `can not resolve package.json file path`

### CODE REVIEW

**Comment by OleksandrKucherenko** [comment]
@janicduplessis @charpeni ^

**Comment by OleksandrKucherenko** [comment]
@janicduplessis - how much time it can take to merge small improvements? No push, but will be great to have some timeline

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[*]
```
Comment (Johan-dutoit) [inline .editorconfig]: This file should be excluded

File: android/build.gradle [hunk 2]
```diff
@@ -51,7 +70,8 @@ repositories {
 }
 
 dependencies {
-    implementation "com.facebook.react:react-native:+"  // From node_modules
+    //noinspection GradleDynamicVersion
+    api "com.facebook.react:react-native:+"  // From node_modules
```
Comment (Johan-dutoit) [inline android/build.gradle]: api is deprecated

File: .editorconfig (hunk 1 above)
Comment (OleksandrKucherenko) [reply .editorconfig]: its a high-level configuration of tab/space formatting for many IDEs. for RN its quite common that devs use many IDEs: Intellij, vscode, atom, etc. .editorconfig is the only cross-ide thing that recognized by all of them.

File: android/build.gradle (hunk 2 above)
Comment (OleksandrKucherenko) [reply android/build.gradle]: Consider you have app module which uses lib1 as a library and lib1 uses lib2 as a library. Something like this: app -> lib1 -> lib2.

Now when using api lib2 in lib1, then app can see lib2 codes when using: api lib1 or implementation lib1 in the app module.

BUT when using implementation lib2 in lib1, then app can not see the lib2 codes.

src: https://stackoverflow.com/questions/44413952/gradle-implementation-vs-api-configuration

**Pull request closed without merging**